File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    dic = dict()
    answer = 0
    for idx in range(len(want)):
        dic[want[idx]] = number[idx]

    for i in range(len(discount) - 9):
        wantDic = dic.copy()
        
        for j in range(i, i + 10):
            if discount[j] in wantDic.keys() and wantDic[discount[j]] > 0:
                wantDic[discount[j]] -= 1
        if sum(wantDic.values()) <= 0:
            answer+=1

    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    const purchaseList = {}
    
    want.forEach((val, idx) => {
        purchaseList[val] = number[idx]
    })
    
    for(let i = 0; i < discount.length; i++) {
        const purchaseListCopy = {...purchaseList}
        for(let j = i; j < i+10; j++) {
            if(discount.length < j) break;
            if(purchaseListCopy[discount[j]]) {
                purchaseListCopy[discount[j]] -= 1
            }
        }
        if(!Object.values(purchaseListCopy).reduce((a, b) => a + b)) {
            answer+=1
        }
    }
    return answer
}